src/racer/core.rs

```diff
@@ -371,7 +372,7 @@ impl fmt::Debug for PathSearch {
 pub struct IndexedSource {
     pub code: String,
     pub idx: Vec<SourceByteRange>,
-    pub lines: RefCell<Vec<SourceByteRange>>
+    pub lines: Arc<RwLock<Vec<SourceByteRange>>>,
```

Comment:
Why is the `Arc` needed here?

Reply:
Good point, and something I realized as well.

We really just want lazy initialization here - I'm looking for a suitable crate.

Reply:
What about just having a RwLock around an empty Vec?